app/controllers/api/physical_servers_controller.rb

```diff
@@ -52,6 +52,8 @@ def change_resource_state(state, type, id)
           api_log_info(desc)
           task_id = queue_object_action(server, desc, :method_name => state, :role => :ems_operations)
           action_result(true, desc, :task_id => task_id)
+        rescue ActiveRecord::RecordNotFound => err
+          single_resource? ? raise(err) : action_result(false, err.to_s)
```

Comment:
I'm not sure I understand how this is working. If I understand this correctly, the call to `single_resource?` here should raise a `NoMethodError`

Reply:
@imtayadeway it shouldn't raise any NoMethodError because this method is defined on its hierarchy on file [results.rb](https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/base_controller/results.rb), Besides this there is a spec for this method that confirms that it works well